pdr_backend/aimodel/dash_plots/callbacks.py

```diff
@@ -0,0 +1,262 @@
+        [
+            Input("feed-dropdown", "value"),
+            Input("date-picker-range", "start_date"),
+            Input("date-picker-range", "end_date"),
+            Input("transition-data", "data"),
+            Input("autocorelation-lag", "value"),
+        ],
+        State("file-data", "data"),
+    )
+    def create_autocorelation_plots(
+        feed_data,
+        date_picker_start_date,
+        date_picker_end_date,
+        transition,
+        lag,
+        files_data,
+    ):
+        if transition is None:
+            return dash.no_update
+
+        nlags = int(lag) if lag is not None else 1
+        do_boxcox = transition["BC"]
+        differencing_order = transition["D"]
+
+        files_data[feed_data] = filter_file_data_by_date(
+            files_data[feed_data],
+            datetime.strptime(date_picker_start_date, DATE_STRING_FORMAT),
+            datetime.strptime(date_picker_end_date, DATE_STRING_FORMAT),
+        )
+
+        y = files_data[feed_data]["close_data"]
+
+        # get data for autocorelation
+        y = np.array(y)
+        if do_boxcox:
+            y, _ = stats.boxcox(y)
+
+        for _ in range(differencing_order):
+            y = y[1:] - y[:-1]
+
+        data = AutocorrelationPlotdataFactory.build(y, nlags)
+        transition_text = f" for BC={'T' if do_boxcox else 'F'},D={differencing_order}"
+
+        acf = plot_acf(data)
+        pacf = plot_pacf(data)
+
+        autocorelation_charts = get_column_graphs(
+            [
+                {"fig": acf, "graph_id": "autocorelation"},
+                {"fig": pacf, "graph_id": "pautocorelation"},
+            ],
+            title="Autocorrelation (ACF)" + transition_text,
+            tooltip=AUTOCORRELATION_TOOLTIP,
+        )
+        return autocorelation_charts
+
+    @app.callback(
+        Output("seasonal_column", "children"),
+        [
+            Input("feed-dropdown", "value"),
+            Input("date-picker-range", "start_date"),
+            Input("date-picker-range", "end_date"),
+            Input("transition-data", "data"),
+        ],
+        State("file-data", "data"),
+    )
+    # pylint: disable=unused-argument
+    def create_seasonal_plots(
+        feed_data, date_picker_start_date, date_picker_end_date, transition, files_data
+    ):
+        if transition is None:
+            return dash.no_update
+
+        do_boxcox = transition["BC"]
+        differencing_order = transition["D"]
+
+        files_data[feed_data] = filter_file_data_by_date(
+            files_data[feed_data],
+            datetime.strptime(date_picker_start_date, DATE_STRING_FORMAT),
+            datetime.strptime(date_picker_end_date, DATE_STRING_FORMAT),
+        )
+
+        # get data for seasonal
+        y = files_data[feed_data]["close_data"]
+        y = np.array(y)
+        timestamp_str = files_data[feed_data]["timestamps"][0]
+        st = UnixTimeS.to_milliseconds(timestamp_str.timestamp())
```

Comment:
I don't think this is right. Shouldn't it be UnixTimeS(timestamp_str.timestamp()).to_milliseconds() ?

Reply:
They both return the same result